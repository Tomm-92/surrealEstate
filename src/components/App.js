import React, { useState } from "react";
import "../styles/app.css";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faHouse,
  faBed,
  faBath,
  faEnvelope,
} from "@fortawesome/free-solid-svg-icons";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";

const App = () => {
  const [userID, setUserID] = useState("");
  const handleLogin = (response) => {
    setUserID(response.userID);
  };

  const handleLogout = () => {
    window.FB.logout(setUserID(""));
  };

  return (
    <Router>
      <div className="App">
        <NavBar userID={userID} onLogin={handleLogin} onLogout={handleLogout} />
        <Routes>
          <Route path="/" element={<Properties userID={userID} />} />
          <Route path="/add-property" element={<AddProperty />} />
        </Routes>
      </div>
    </Router>
  );
};

library.add(faHouse, faBath, faBed, faEnvelope);
export default App;
