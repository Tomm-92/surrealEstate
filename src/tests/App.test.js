import { render, screen } from '@testing-library/react';
import App from '../components/App';

test('renders the page header', () => {
  render(<App />);
  const headerElement = screen.getByText(/Surreal Estate/i);
  expect(headerElement).toBeInTheDocument();
});
