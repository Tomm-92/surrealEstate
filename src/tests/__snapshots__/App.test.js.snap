// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navBar"
    >
      <img
        alt="nasa_logo"
        class="surreal-logo"
        src="Logo.png"
      />
      <ul
        class="navbar-links"
      >
        <a
          class="navbar-links-item"
          href="/"
        >
          View Properties
        </a>
        <a
          class="navbar-links-item"
          href="/add-property"
        >
          Add Properties
        </a>
      </ul>
    </div>
    <div>
       Properties Page 
    </div>
  </div>
</DocumentFragment>
`;
